src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,216 @@
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths) => new SpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new SpanND<T>(address, lengths, false, strides);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths) => new ReadOnlySpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new ReadOnlySpanND<T>(address, lengths, false, strides);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateFromEnumerable<T>(IEnumerable<T> data)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            T[] values = data.ToArray();
+            return new Tensor<T>(values, [values.Length], false);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateUniform<T>(params ReadOnlySpan<nint> lengths)
```

Comment:
nit: This name is still very ambiguous (and potentially confusing).

Really this is doing `Create and fill with uniformly distributed random data` and I think we need a name or setup that better captures the 2 operations being done and is more descriptive as to what `Uniform` means here. 

Similar comments exist for the other `Create*` APIs that are doing similar "create and fill with ..." operations

Reply:
How about `CreateAndFillUniform` and `CreateAndFillGaussianNormal`?